core/templates/pages/exploration-player-page/services/question-player-engine.service.ts

```diff
@@ -96,34 +87,35 @@ export class QuestionPlayerEngineService {
       successCallback: (initialCard: StateCard, nextFocusLabel: string) => void,
       errorCallback: () => void): void {
     if (!this.questions || this.questions.length === 0) {
+      this.alertsService.addWarning('There are no questions to display.');
```

Comment:
Optional: Thanks for adding this! But can we move the `if` block before this function is called? It seems like question list being non-empty is a prereq for loading initial question.

Reply:
I think we can't directly move the if block since we are calling error call back here.

Reply:
Then errorCallback need not be passed into this function and it should be handled outside?

Reply:
done!